java/flight/flight-jdbc/src/main/java/org/apache/arrow/jdbc/Driver.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.jdbc;
+
+import java.sql.Connection;
+import java.sql.DriverPropertyInfo;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import org.slf4j.LoggerFactory;
+
+/**
+ * Driver.
+ */
+public class Driver implements java.sql.Driver {
+
+  private static final org.slf4j.Logger logger = LoggerFactory.getLogger(Driver.class);
+
+  /** JDBC connection string prefix. */
+  private static final String PREFIX = "jdbc:arrow://";
+
+  @Override
+  public Connection connect(String url, Properties properties) throws SQLException {
+    logger.info("connect() url={}", url);
+    //TODO this needs much more work to parse full URLs but this is enough to get end to end tests running
```

Comment:
[`java.net.URI`](https://docs.oracle.com/javase/7/docs/api/java/net/URI.html) should be able to do all the parsing.